### PR TITLE
merge bitcoin#18914, #13306, #16424, #13899, #17486, #17880, #18145, #18843, #16710: split warnings out of CXXFLAGS, add more flags

### PR DESCRIPTION
**Pull request by kittywhiskers**
Should take care of this warning generated on macOS Mojave

```
In file included from qt/moc_signverifymessagedialog.cpp:8:
./qt/signverifymessagedialog.h:36:10: warning: 'eventFilter' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
    bool eventFilter(QObject *object, QEvent *event);
         ^
/Users/localhost/Projects/dash/depends/x86_64-apple-darwin18.7.0/include/QtWidgets/qdialog.h:111:10: note: overridden virtual function is here
    bool eventFilter(QObject *, QEvent *) override;
         ^
```

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Better to backport #18914 to avoid merge conflicts

File: src/Makefile.leveldb.include [hunk 1]
```diff
@@ -30,7 +30,7 @@ LEVELDB_CPPFLAGS_INT += -DLEVELDB_PLATFORM_POSIX
 endif
 
 leveldb_libleveldb_a_CPPFLAGS = $(AM_CPPFLAGS) $(LEVELDB_CPPFLAGS_INT) $(LEVELDB_CPPFLAGS)
-leveldb_libleveldb_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+leveldb_libleveldb_a_CXXFLAGS = -Wsuggest-override -Werror=suggest-override $(AM_CXXFLAGS) $(PIE_FLAGS)
```
Comment (UdjinM6) [inline src/Makefile.leveldb.include:33]: should backport 18843 first

File: configure.ac [hunk 2]
```diff
@@ -382,6 +386,8 @@ if test "x$CXXFLAGS_overridden" = "xno"; then
   AX_CHECK_COMPILE_FLAG([-Wunused-local-typedef],[CXXFLAGS="$CXXFLAGS -Wno-unused-local-typedef"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wdeprecated-register],[CXXFLAGS="$CXXFLAGS -Wno-deprecated-register"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wimplicit-fallthrough],[CXXFLAGS="$CXXFLAGS -Wno-implicit-fallthrough"],,[[$CXXFLAG_WERROR]])
+  AX_CHECK_COMPILE_FLAG([-Wsuggest-override],[WARN_CXXFLAGS="$WARN_CXXFLAGS -Wsuggest-override"],,[[$CXXFLAG_WERROR]],
+                        [AC_LANG_SOURCE([[struct A { virtual void f(); }; struct B : A { void f() final; };]])])
```
Comment (UdjinM6) [inline configure.ac:390]: this belongs to another block, should be at line 380-ish

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
```diff
diff --git a/src/qt/qrdialog.h b/src/qt/qrdialog.h
index 46d9cbcb1a..3c503fe1e3 100644
--- a/src/qt/qrdialog.h
+++ b/src/qt/qrdialog.h
@@ -35,8 +35,8 @@ public Q_SLOTS:
     void copyImage();
 
 protected:
-    virtual void mousePressEvent(QMouseEvent *event);
-    virtual void contextMenuEvent(QContextMenuEvent *event);
+    virtual void mousePressEvent(QMouseEvent *event) override;
+    virtual void contextMenuEvent(QContextMenuEvent *event) override;
 
 private:
     QMenu *contextMenu;
```

File: configure.ac (hunk 2 above)
Comment (kittywhiskers) [reply configure.ac]: Resolved 🙂

File: src/Makefile.leveldb.include (hunk 1 above)
Comment (kittywhiskers) [reply src/Makefile.leveldb.include]: Backported #18843 and then some more, resolved!

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: configure.ac [hunk 3]
```diff
@@ -1572,11 +1574,11 @@ echo
...
-echo "  CXXFLAGS            = $DEBUG_CXXFLAGS $HARDENED_CXXFLAGS $ERROR_CXXFLAGS $GPROF_CXXFLAGS $CXXFLAGS"
-echo "  LDFLAGS             = $PTHREAD_CFLAGS $HARDENED_LDFLAGS $GPROF_LDFLAGS $LDFLAGS"
-echo "  ARFLAGS             = $ARFLAGS"
+echo "  CC            = $CC"
+echo "  CFLAGS        = $CFLAGS"
+echo "  CPPFLAGS      = $DEBUG_CPPFLAGS $HARDENED_CPPFLAGS $CPPFLAGS"
+echo "  CXX           = $CXX"
+echo "  CXXFLAGS      = $DEBUG_CXXFLAGS $HARDENED_CXXFLAGS $WARN_CXXFLAGS $NOWARN_CXXFLAGS $ERROR_CXXFLAGS $GPROF_CXXFLAGS $CXXFLAGS"
+echo "  LDFLAGS       = $PTHREAD_CFLAGS $HARDENED_LDFLAGS $GPROF_LDFLAGS $LDFLAGS"
+echo "  ARFLAGS       = $ARFLAGS"
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply configure.ac:1599]: Bitcoin Core doesn't have those whitespaces, so should we prefer Bitcoin's whitespacing or retain Dash's?

[the remark this answers is not in the document]
Comment (UdjinM6) [reply configure.ac:1599]: should keep the output aligned and we have some wider (text length wide) options like `stacktraces enabled` above

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
13899 is incomplete but it's fine, can be fixed later (there are quite a few other things to fix when compiled with these new flags)

ACK

**Pull request merged**